### PR TITLE
Update FIPS documentation

### PR DESCRIPTION
**Pull request by cjellick**
#### Proposed Changes ####
Update fips documentation for the following changes:
- Explain ingress's new fips compliance.
- Link to Rancher's CMVP certificate.
- Call out that only the canal cni is fips compliant

#### Types of Changes ####
Just a docs change

#### Verification ####
N/A, just docs

#### Linked Issues ####
- For ingress bit: https://github.com/rancher/rke2/issues/1511
- For the CMVP bit, I'm submitting without an issue
- For the CNI bit: https://github.com/rancher/rke2/issues/1180

### CODE REVIEW

File: docs/security/fips_support.md [hunk 1]
```diff
@@ -43,6 +41,17 @@ To ensure that all aspects of the system architecture are using FIPS 140-2 compl
...
+
+As of v1.21.2, RKE2 supports selecting a different CNI via the `--cni` flag and comes bundled with several CNIs including Canal (default), Calico, Cilium, and Multus. Of these, only Canal (the default) is rebuilt for FIPS compliance.
+
 ## Ingress
 
-The NGINX Ingress included with RKE2 is **not** currently FIPS enabled. It can, however, be [disabled and replaced](../advanced.md#disabling-server-charts) by the cluster operator/owner.
+RKE2 ships with NGNIX as its default ingress provider. As of v1.21+, this component is FIPS compliant. There are two primary sub-components for NGINX ingress:
+
+- controller - responsible for monitoring Kubernetes and configuring NGINX accordingly
+- NGINX server- responsible for handling and routing traffic
```
Comment (brandond) [inline docs/security/fips_support.md:53]:
```suggestion
- server - responsible for accepting and routing traffic
```

File: docs/security/fips_support.md [hunk 2]
```diff
@@ -43,6 +41,17 @@ To ensure that all aspects of the system architecture are using FIPS 140-2 compl
...
+
+The controller is written in Go and as such is compiled using our [FIPS compatible Go compiler](./fips_support.md#use-of-fips-compatible-go-compiler).
+
+The NGINX server is written in C and also requires OpenSSL to function properly. As such, it leverages a FIPS-validated version of OpenSSL to achieve FIPS compliance.
```
Comment (brandond) [inline docs/security/fips_support.md:57]:
```suggestion
The server is written in C and also requires OpenSSL to function properly. As such, it leverages a FIPS-validated version of OpenSSL to achieve FIPS compliance.
```

**Review by brandond — CHANGES_REQUESTED** [review]
couple nits

File: docs/security/fips_support.md (hunk 1 above)
Comment (cjellick) [reply docs/security/fips_support.md]: 👍

File: docs/security/fips_support.md (hunk 2 above)
Comment (cjellick) [reply docs/security/fips_support.md]: 👍

File: docs/security/fips_support.md [hunk 3]
```diff
@@ -43,6 +41,17 @@ To ensure that all aspects of the system architecture are using FIPS 140-2 compl
...
+## CNI
+
+As of v1.21.2, RKE2 supports selecting a different CNI via the `--cni` flag and comes bundled with several CNIs including Canal (default), Calico, Cilium, and Multus. Of these, only Canal (the default) is rebuilt for FIPS compliance.
+
 ## Ingress
 
-The NGINX Ingress included with RKE2 is **not** currently FIPS enabled. It can, however, be [disabled and replaced](../advanced.md#disabling-server-charts) by the cluster operator/owner.
+RKE2 ships with NGNIX as its default ingress provider. As of v1.21+, this component is FIPS compliant. There are two primary sub-components for NGINX ingress:
+
+- controller - responsible for monitoring Kubernetes and configuring NGINX accordingly
```
Comment (brandond) [inline docs/security/fips_support.md:52]:
```suggestion
- controller - responsible for monitoring/updating Kubernetes resources and configuring the server accordingly
```

**Review by brandond — CHANGES_REQUESTED** [review]
one more minor nit

File: docs/security/fips_support.md (hunk 3 above)
Comment (cjellick) [reply docs/security/fips_support.md]: 👍

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**